lib/translator/status.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIB_TRANSLATOR_STATUS_H
+#define LIB_TRANSLATOR_STATUS_H
+
+#include "lib/scsi_defs.h"
+#include "third_party/spdk_defs/nvme_defs.h"
+
+namespace translator {
+
+struct ScsiStatus {
+  scsi_defs::Status status;
+  scsi_defs::SenseKey sense_key;
+  scsi_defs::AdditionalSenseCode asc;
+  scsi_defs::AdditionalSenseCodeQualifier ascq;
+
+  bool operator==(const ScsiStatus& other) const;
+};
+
+constexpr ScsiStatus kDefaultScsiStatus = ScsiStatus{
+    .status = scsi_defs::Status::kCheckCondition,
+    .sense_key = scsi_defs::SenseKey::kNoSense,
+    .asc = scsi_defs::AdditionalSenseCode::kNoAdditionalSenseInfo,
+    .ascq = scsi_defs::AdditionalSenseCodeQualifier::kNoAdditionalSenseInfo,
+};
```

Comment:
`constexpr`s appearing in headers should be marked `inline` or else they're a severe correctness hazard: if they're not `inline` then any use of them inside an `inline` function that gets compiled into multiple compilation unit is undefined behavior!

Do we actually need this to be in the header (i.e. visible to other compilation units) at all, though?

Reply:
I was thinking they might be useful if any function wants to check whether the `ScsiStatus` is just the default value. Although, we could keep this just in the .cc for now and expand its visibility when the need arises.